ksql-rest-app/src/main/java/io/confluent/ksql/rest/util/ClusterTerminator.java

```diff
@@ -33,26 +32,27 @@
 
 public class ClusterTerminator {
 
-  private final KsqlConfig ksqlConfig;
   private final KsqlEngine ksqlEngine;
   private final ServiceContext serviceContext;
+  private final List<String> managedTopics;
 
   public ClusterTerminator(
       final KsqlConfig ksqlConfig,
       final KsqlEngine ksqlEngine,
-      final ServiceContext serviceContext
+      final ServiceContext serviceContext,
+      final List<String> managedTopics
   ) {
     Objects.requireNonNull(ksqlConfig, "ksqlConfig is null.");
     Objects.requireNonNull(ksqlEngine, "ksqlEngine is null.");
-    this.ksqlConfig = ksqlConfig;
     this.ksqlEngine = ksqlEngine;
     this.serviceContext = Objects.requireNonNull(serviceContext);
+    this.managedTopics = ImmutableList.copyOf(Objects.requireNonNull(managedTopics));
   }
 
   public void terminateCluster(final List<String> deleteTopicPatterns) {
     terminatePersistentQueries();
     deleteSinkTopics(deleteTopicPatterns);
-    deleteCommandTopic();
+    deleteTopics(managedTopics);
```

Comment:
I'm wondering if we should close the `ksqlEngine` even if we fail to delete the topics. Thoughts?

Reply:
I think this should either be retry-able if it fails, or the command-runner should keep retrying on success with some polling mechanism in the API. It doesn't look like it works this way atm (it'll just leave the cluster in a partially torn down state), but we should fix that. That would be out-of-scope for this PR though.